### PR TITLE
Fix issue about isDomainPointedToThisFDM and adding whitelisting feature and remove mandatoryApps

### PR DESCRIPTION
**Pull request by fegauthier**
- Adding app whitelisting
- Adding app owners
- Fix issue about not providing IP for isDomainPointedToThisFDM method
- Remove useless await
- Moving mandatory apps to config file

### CODE REVIEW

File: src/services/domainService.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const { getCustomConfigs } = require('./application/custom');
 let myIP = null;
 let myFDMnameORip = null;
 
-const mandatoryApps = ['explorer', 'KDLaunch', 'website', 'Kadena3', 'Kadena4', 'HavenNodeMainnet'];
+const mandatoryApps = [];
```
Comment (FluxST) [inline src/services/domainService.js:15]: Why are we removing the mandatory apps here? Are these not mandatory anymore?

Comment (FluxST) [reply src/services/domainService.js]: @TheTrunk actually do you know why we have those mandatory apps?

@fegauthier, good idea, Ill add mandatoryApps to config file once we verify that we still need them

File: src/services/domainService.js [hunk 2]
```diff
@@ -70,7 +68,13 @@ async function createSSLDirectory() {
 async function generateAndReplaceMainApplicationHaproxyConfig() {
   try {
     // get applications on the network
-    const applicationSpecifications = await fluxService.getAppSpecifications();
+    let applicationSpecifications = await fluxService.getAppSpecifications();
+
+    // If there's ownersApps, only include them
+    if(config.ownersApps.length > 0) {
```
Comment (TheTrunk) [inline src/services/domainService.js:74]: we can do just 'config.ownersApps.length'
+ space for lint

**Review by TheTrunk — APPROVED** [review]
Nice addition 👍

File: src/services/domainService.js (hunk 1 above)
Comment (TheTrunk) [reply src/services/domainService.js]: Yes mandatory apps are needed as we can get a bad node from fluxos feeding this fdm. Or some app expired on flux that should not expire but some instnace still may be running - in this case we are getting some notifications that fdm failed to reload and we can quickly fix that.

File: src/services/domainService.js (hunk 2 above)
Comment (fegauthier) [reply src/services/domainService.js]: Done! Thanks!

**Pull request merged**